common/src/main/java/com/wynntils/core/persisted/config/ConfigManager.java

```diff
@@ -262,4 +266,61 @@ public Stream<Config<?>> getConfigsForOwner(PersistedOwner owner) {
         return getConfigs()
                 .filter(config -> Managers.Persisted.getMetadata(config).owner() == owner);
     }
+
+    public boolean importConfig(String jsonInput, List<Configurable> configsToImport) {
+        try {
+            Map<String, Object> configData =
+                    Managers.Json.GSON.fromJson(jsonInput, new TypeToken<HashMap<String, Object>>() {}.getType());
+
+            // Loop through all features chosen to import to
+            for (Configurable feature : configsToImport) {
+                // Loop through the visible configs only as they are the only configs to be imported
+                for (Config<?> configOption : feature.getVisibleConfigOptions()) {
+                    String configOptionName = configOption.getJsonName();
+
+                    // If the config data contains this config option, then it can be imported
+                    if (configData.containsKey(configOptionName)) {
+                        Object configOptionValue = configData.get(configOptionName);
+                        setConfigValue(configOption, configOptionValue);
+                    }
+                }
+            }
+
+            return true;
+        } catch (JsonSyntaxException ex) {
```

Comment:
Isn't there more types of exception that can be thrown in the code above? If `setConfigValue` fails to due to type miss-match, or some other issue, we need to catch that as well.

Reply:
`tryParseStringValue` catches a generic exception if it fails which then won't call `setValue` so I think it should be fine? I have now added a check for configData being null though as that wasn't checked

Reply:
Yes, you are right.